tests/main/try/task.yaml

```diff
@@ -0,0 +1,65 @@
+summary: Check that try command works
+
+environment:
+  PORT: 8081
+  SERVICE_FILE: "./service.sh"
+  SERVICE_NAME: "test-service"
+
+prepare: |
+  echo "Given a service listening on a port"
+  echo "#!/bin/sh\nwhile true; do echo \"HTTP/1.1 200 OK\n\nok\n\" |  nc -l -p $PORT -q 1; done" > $SERVICE_FILE
+  chmod a+x $SERVICE_FILE
+  systemd-run --unit $SERVICE_NAME $SERVICE_FILE
+  while ! netstat -lnt | grep -Pq "tcp.*?:$PORT +.*?LISTEN\n*"; do sleep 0.5; done
+
+restore: |
+  systemctl stop $SERVICE_NAME
+  rm -f $SERVICE_FILE
+
+execute: |
+  echo "Given a buildable snap in a known directory"
+  echo "When try is executed on that directory"
+  snap try ../../lib/snaps/basic-binaries
+
+  echo "Then the snap is listed as installed with try in the notes"
+  expected="(?s)Name +Version +Rev +Developer +Notes\n\
+  basic-binaries +.*?try"
+  echo "$(snap list)" | grep -Pzq "$expected"
+
+  echo "And commands from the snap-try binary can be run"
+  basic-binaries.success
+
+  echo "And commands from the snap-try binary can read in their own dir"
+  echo "$(basic-binaries.cat)" | grep -Pzq "I output myself"
+
+  echo "====================================="
+
+  echo "Given a buildable snap which access confinement-protected resources in a known directory"
+  echo "When try is executed on that directory"
+  snap try ../../lib/snaps/basic-binaries
+
+  echo "Then the snap command is not able to access the protected resource"
+  if basic-binaries.disallowed; then
+    echo "Expected confinement denial in try mode not worked" && exit 1
+  fi
+
+  echo "====================================="
+
+  echo "Given a buildable snap which access confinement-protected resources in a known directory"
+  echo "When try is executed on that directory with devmode enabled"
+  snap try ../../lib/snaps/basic-binaries --devmode
+
+  echo "Then the snap command is able to access the protected resource"
+  result=$(basic-binaries.disallowed)
+  [ ${#result} -ge 0 ]
```

Comment:
If we just call the application directly without anything else, it will already test its execution naturally:
```
basic-binaries.disallowed
```